### PR TITLE
Add support for maat in Firefox 49

### PR DESCRIPTION
**Pull request by mjneil**
## Description
The newest version of Firefox (49) fixed a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1247138) that prevented Firefox from supporting audio configuration change during playback. This required blacklisting of alternate audio tracks. This PR adds a check for Firefox 49 using feature detection to allow change in audio info for version 49, but continues to blacklist in versions 48 and below.

## Specific Changes proposed
Add an additional condition to check for the existence of the property `window.isSecureContext`, which was added to Firefox 49.

## Requirements Checklist
- [ ] Feature implemented / Bug fixed
- [ ] Unit Tests updated or fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: test/videojs-contrib-hls.test.js [hunk 1]
```diff
@@ -1792,11 +1792,15 @@ QUnit.test('adds audio tracks if we have parsed some from a playlist', function(
   QUnit.equal(vjsAudioTracks[0].enabled, false, 'main track is disabled');
 });
 
-QUnit.test('when audioinfo changes on an independent audio track in Firefox, the enabled track is blacklisted and removed', function() {
+QUnit.test('when audioinfo changes on an independent audio track in Firefox 48 & below, the enabled track is blacklisted and removed', function() {
```
Comment (dmlap) [inline test/videojs-contrib-hls.test.js]: I can't wait to delete these suckers.

File: src/videojs-contrib-hls.js [hunk 2]
```diff
@@ -244,6 +244,18 @@ Hls.isSupported = function() {
...
 
+/**
+ * Determines whether the browser supports a change in the audio configuration
+ * during playback. Currently only Firefox 48 and below do not support this.
+ * window.isSecureContext is a propterty that was added to window in firefox 49,
+ * so we can use it to detect Firefox 49+.
+ *
+ * @return {Boolean} Whether the browser supports audio config change during playback
+ */
+Hls.supportsAudioInfoChange = function() {
```
Comment (dmlap) [inline src/videojs-contrib-hls.js]: We should probably make this property private so we can remove it once FF 48 falls out of support.

Comment (mjneil) [inline src/videojs-contrib-hls.js]: Made private

File: src/videojs-contrib-hls.js [hunk 3]
```diff
@@ -244,6 +244,18 @@ Hls.isSupported = function() {
...
+/**
+ * Determines whether the browser supports a change in the audio configuration
+ * during playback. Currently only Firefox 48 and below do not support this.
+ * window.isSecureContext is a propterty that was added to window in firefox 49,
+ * so we can use it to detect Firefox 49+.
+ *
+ * @return {Boolean} Whether the browser supports audio config change during playback
+ */
+Hls.supportsAudioInfoChange_ = function() {
+  return !videojs.browser.IS_FIREFOX || window.hasOwnProperty('isSecureContext');
```
Comment (imbcmdth) [inline src/videojs-contrib-hls.js]: I think we settled on using the UA string parsing to check the browser version.

Comment (mjneil) [inline src/videojs-contrib-hls.js]: done

**Comment by imbcmdth** [comment]
LGTM

**Comment by forbesjo** [comment]
LGTM

**Pull request merged**